File: app/src/main/java/com/example/calculator/adpters/HistoryAdapter.java
package com.example.calculator.adpters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.recyclerview.widget.RecyclerView;
import com.example.calculator.Model.History;
import com.example.calculator.R;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;

public class HistoryAdapter extends RecyclerView.Adapter<HistoryAdapter.ViewHolder> {
    private final Context context;
    private final ArrayList<History> historyItems;

    public HistoryAdapter(Context context2, ArrayList<History> historyItems2) {
        this.context = context2;
        this.historyItems = historyItems2;
    }

    @NotNull
    public ViewHolder onCreateViewHolder(@NotNull ViewGroup parent, int viewType) {
        return new ViewHolder(LayoutInflater.from(this.context).inflate(R.layout.history_item, parent, false));
    }

    public void onBindViewHolder(ViewHolder holder, int position) {
        History history = this.historyItems.get(position);
        holder.expression.setText(history.getExpr());
        holder.answer.setText(history.getAnswer());
    }

    public int getItemCount() {
        return this.historyItems.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        TextView answer;
        TextView expression;

        public ViewHolder(View itemView) {
            super(itemView);
            this.expression = itemView.findViewById(R.id.expression);
            this.answer = itemView.findViewById(R.id.answer);
        }
    }
}

File: app/src/main/java/com/inator/calculator/adpters/HistoryAdapter.kt
package com.inator.calculator.adpters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.inator.calculator.History.History
import com.inator.calculator.R
import java.util.*

class HistoryAdapter(private val context: Context, private val historyItems: ArrayList<History>) :
    RecyclerView.Adapter<HistoryAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.history_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val history = historyItems[position]
        holder.expression.text = history.expr
        holder.answer.text = history.answer
    }

    override fun getItemCount(): Int {
        return historyItems.size
    }

    fun updateList(list: List<History>) {
        historyItems.clear()
        historyItems.addAll(list)
        notifyDataSetChanged()
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView),
        View.OnClickListener {
        val answer: TextView = itemView.findViewById(R.id.answer)
        val expression: TextView = itemView.findViewById(R.id.expression)
        override fun onClick(v: View) {
//            CalculatorFragment.draggablePanel?.smoothPanelClose(300)
//            val calculatorFragment =
//                ((context as AppCompatActivity?)!!.supportFragmentManager.findFragmentById(R.id.historyContainer) as CalculatorFragment?)!!
//            calculatorFragment.setInputText(expression.text.toString())
        }

        init {
            itemView.setOnClickListener(this)
        }
    }
}